### PR TITLE
prune --max-unused does not limit metadata repacking

### PR DESCRIPTION
**Pull request by cyounkins**
What does this PR change? What problem does it solve?
-----------------------------------------------------

Adding minor clarification to documentation of `prune --max-unused` to indicate it does not limit metadata repacking. The referenced PR indicates `max-repack-size` CAN limit metadata repacking.

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

https://forum.restic.net/t/max-unused-unlimited-still-repacks/3661
https://github.com/restic/restic/pull/3228

Checklist
---------

- [x] I have read the [Contribution Guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches)
- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [ ] I have added tests for all changes in this PR
- [ ] I have added documentation for the changes (in the manual)
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/restic/blob/master/changelog/TEMPLATE))
- [ ] I have run `gofmt` on the code in all commits
- [ ] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: doc/060_forget.rst [hunk 1]
```diff
@@ -344,7 +344,7 @@ The ``prune`` command accepts the following options:
       operation.
 
    Restic tries to repack as little data as possible while still ensuring this 
-   limit for unused data.
+   limit for unused data. Note that this option does not limit metadata repacking.
```
Comment (MichaelEischer) [inline doc/060_forget.rst:347]: In my opinion this remark should be part of the explanation of `unlimited`. And I'd reword it a bit to something like `Note that metadata is always repacked`.

When using `--max-unused` with a specific limit then it probably doesn't matter much that the target for unused metadata is always zero. That only becomes obvious when using the unlimited option.

Comment (cyounkins) [reply doc/060_forget.rst:347]: I was confused but understand a bit better now.

Is metadata repacking limited by the `--max-repack-size` option? In my referenced forum post I used `--max-unused unlimited` and it repacked ~430 MB. If I had instead set `--max-repack-size 200M`, how much would it have repacked?

Comment (MichaelEischer) [reply doc/060_forget.rst:347]: `--max-repack-size` limits metadata repacking. So it should have been about 200MB. prune just repacks all metadata first before repacking any data files.

Comment (cyounkins) [reply doc/060_forget.rst]: So since `--max-repack-size` limits metadata repacking, I think it's confusing to say "metadata is always repacked" even in the context of the `unlimited` option of `--max-unused`. So I'm going to put "Note that metadata will still be repacked" in the context of the `unlimited` option.

If you disagree, feel free to commandeer or tell me to just put, "Note that metadata is always repacked". Thanks!

Comment (MichaelEischer) [reply doc/060_forget.rst]: I actually prefer your wording over my suggestion, so lets keep that.

**Comment by MichaelEischer** [comment]
Could you squash the commits into one? Two commits are a bit overkill for changing a single line ;-) .

**Comment by cyounkins** [comment]
Oops! This is what happens when I try to use the GitHub interface for a simple edit. Fixed now.

**Review by MichaelEischer — APPROVED** [review]
LGTM. Thanks!
